### ISSUE TITLE
Can't this be done using |> instead?

### ISSUE THREAD

**Issue by imranismail** [miraks/babel-plugin-pipe-operator#1]

**Comment by miraks** [comment]
Unfortunately, no. Constructions like this

``` javascript
a |> b
```

will give syntax error in JavaScript. It will be possible if Babel team provide API for parser plugins, but I don't know if they're planning to do that.

**Comment by Swizz** [comment]
To avoid syntax error, Babel seem to use two kinds of plugins : `syntax` and `transform`.

I think `syntax` allow the parser to work well with non standard syntax.
And `transform`, made the new syntax to JS one.

[More informations.](https://github.com/thejameskyle/babel-handbook/blob/master/translations/en/plugin-handbook.md)

**Comment by miraks** [comment]
Yes, but API for parser (syntax) plugins isn't public and really poor. Actually babels "official" syntax plugins are hardcoded into [babylon](https://github.com/babel/babylon). For example, function bind syntax plugin: [plugin code](https://github.com/babel/babel/blob/4c371132ae7321f6d08567eab54a59049e07f246/packages/babel-plugin-syntax-function-bind/src/index.js#L4) and [babylon check](https://github.com/babel/babylon/blob/8b150813f5a534e4279df25c41c37cc9dd860ebe/src/tokenizer/index.js#L412).

**Comment by Swizz** [comment]
Oh. 
I am just starting with Babel plugins.

So, the best way is to play with [mrapogee/babel-plugin-syntax-pipeline-operator](/mrapogee/babel-plugin-syntax-pipeline-operator) and [mrapogee/babel-plugin-transform-pipeline-operator](/mrapogee/babel-plugin-transform-pipeline-operator) and to made a PR on [babel/babylon](/babel/babylon) to work well with the `pipelineOperator` option ?

Someone is interested about this functionality or this one work well for your needs ?

_( cc @mrapogee )_

**Comment by miraks** [comment]
Looks like there's already been [PR](https://github.com/babel/babel/pull/3159) from @mrapogee but it never was merged.

**Comment by Swizz** [comment]
Oh. Got it ! 
Thanks for all your answers.

**Comment by samdesota** [comment]
Hey all,

I was playing around with this a few months ago and I forked babel to get this working on my setup, but the PR and code referenced above is actually from @mindeavor

I did make babylon branch [mrapogee/babylon/tree/feature-pipeline](https://github.com/mrapogee/babylon/tree/feature-pipeline) that adds the pipeline operator to the syntax. If you get babel-core to import the source from that branch, you may be able to compile the pipeline operator if you include babel-plugin-transform-pipeline-operator in your .babelrc, though I haven't fully tested.

**Comment by Swizz** [comment]
I want to keep my hands on the official babylon.

This plugin works well, and finally the `|` operator is sweet. 
I do not need bitwise operators in javascript.

---

But.. In the esnext proposal about pipeline, @mindeavor works with currying.

So ...

``` javascript
let result = "hello"
  |> doubleSay
```

become

``` javascript
let result = doubleSay("hello")
```

---

Here, you choose to unshift it in the arguments array like Elm.

So ...

``` javascript
let result = "hello"
  |> doubleSay(null)
```

become

``` javascript
let result = doubleSay("hello", null)
```

---

I think currying is a better way to do that stuff.
I will work on an alternative which implements currying stuffs.
